File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> set;
        int left=0;
        int max_size=0;
        for(int i=0;i<s.size();i++){
            if(set.count(s[i])){
            while(set.count(s[i])){
                set.erase(s[left]);
                left++;
            }
              
            }
            
            set.insert(s[i]);
            max_size=max(max_size,(int)set.size());
        }
        return max_size;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, stri):
        """
        :type s: str
        :rtype: int
        """
        if stri=="":
            return 0


        s=0
        e=0
        ans=0
        st=set()
        for c in stri:
            if c in st:
                while c in st:
                    
                    st.remove(stri[s])
                    s+=1
            st.add(c)        
            ans=max(ans,(e-s))
            
            e+=1
        return ans+1
        